pkg/logql/log/pipeline.go

```diff
@@ -82,18 +84,21 @@ func IsNoopPipeline(p Pipeline) bool {
 }
 
 type noopStreamPipeline struct {
-	LabelsResult
+	builder *LabelsBuilder
 }
 
-func (n noopStreamPipeline) Process(_ int64, line []byte) ([]byte, LabelsResult, bool) {
-	return line, n.LabelsResult, true
+func (n noopStreamPipeline) Process(_ int64, line []byte, nonIndexedLabels ...labels.Label) ([]byte, LabelsResult, bool) {
+	n.builder.Reset()
+	n.builder.Add(nonIndexedLabels...)
+	return line, n.builder.LabelsResult(), true
 }
 
-func (n noopStreamPipeline) ProcessString(_ int64, line string) (string, LabelsResult, bool) {
-	return line, n.LabelsResult, true
+func (n noopStreamPipeline) ProcessString(ts int64, line string, nonIndexedLabels ...labels.Label) (string, LabelsResult, bool) {
+	lb, lr, ok := n.Process(ts, unsafeGetBytes(line), nonIndexedLabels...)
+	return string(lb), lr, ok
```

Comment:
I think it would be better to do this to avoid some unnecessary allocations
```suggestion
	_, lr, ok := n.Process(ts, unsafeGetBytes(line), nonIndexedLabels...)
	return line, lr, ok
```